### ISSUE TITLE
Refactor Test Cases

### ISSUE THREAD

**Issue by ChiragGhanshani** [project-renard/curie#58]

**Comment by zmughal** [comment]
One possible way is to do inversion of control via a coderef. Remind me to show you an example.

**Comment by zmughal** [comment]
Here is an example:

``` perl
sub build_test {
  my ($cb) = @_;
  return sub {
    my $app = ...;

    $cb->( $app );

    $app->window->destroy;
  }
}

subtest "My test" => build_test( sub {
  my ($app) = @_;
  ...
} );
```

**Comment by ChiragGhanshani** [comment]
Hmmm. I might need you to explain this further in tomorrow's meeting

**Comment by zmughal** [comment]
Sure, we can do that.

**Comment by zmughal** [comment]
Closed by #62.
